src/Symfony/Component/Translation/Provider/ProviderInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Provider;
+
+use Symfony\Component\Translation\TranslatorBagInterface;
+
+interface ProviderInterface
+{
+    public function __toString(): string;
+
+    public function getName(): string;
```

Comment:
Let's remove this method. We had such methods in the past and we have removed them as soon as PHP got the possibility to use `SomeClass::class` notation. We should use the FQCN instead.

Reply:
As all providers are [decorated with FilteringProvider](https://github.com/symfony/symfony/pull/38475/files#diff-dc84272aa7d82c0f559a79e35dd01d0a5f60f48e7ca4ee58fe58558edde7d83fR53), we could not use something like `get_class($provider)` instead of `getName` [here](https://github.com/symfony/symfony/pull/38475/files#diff-1f37873a32276d9edc3138716630ba8505dcc29b12930327d45caa000b8e3c47R131). 

But `getName` could return `self::class`, before 6.0 and PHP 8 supports to use `$provider::class`?

Reply:
Discussed with @nicolas-grekas, `__toString` is not used anymore. So we can remove `getName` and use `__toString` instead.